aten/src/ATen/native/Loss.cpp

```diff
@@ -31,7 +31,7 @@ Tensor cosine_embedding_loss(const Tensor& input1, const Tensor& input2, const T
   auto denom = (mag_square1 * mag_square2).sqrt_();
   auto cos = prod_sum / denom;
 
-  auto zeros = at::zeros_like(target);
+  auto zeros = at::zeros_like(cos);
```

Comment:
can both inputs be integral? margin is a double so that would make neg a double (below), used in output_neg

Reply:
Yes, I think the way we should go with cosine_embedding_loss and hinge_embedding_loss is something like:
```
auto zeros = at::zeros_like(pos);
auto output_pos = at::where(target == 1, pos, zeros);
auto output_neg = at::where(target == -1, neg, zeros.**to(neg.dtype())**);
```

Reply:
I don't see how the inputs being integral matter:
```
auto mag_square1 = (input1 * input1).sum(1) + EPSILON;
auto mag_square2 = (input2 * input2).sum(1) + EPSILON;
```

this line will make everything floating point anyway.

Of course, you should just test this.

Reply:
Added test